File: security/src/main/java/ru/virgil/security/SecurityProperties.java
package ru.virgil.security;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Data
@ConfigurationProperties(prefix = "security")
public class SecurityProperties {

    /**
     * По этим путям можно будет заходить анонимно
     */
    private String[] anonymousPaths = new String[0];
    /**
     * Использовать для сессий заголовок X-Auth-Token вместо Cookies
     */
    private boolean useXAuthToken = false;
    /**
     * Отключить редиректы после успешной авторизации
     */
    private boolean disablePostAuthRedirect = false;
}

File: security/src/main/java/ru/virgil/security/SecurityProperties.kt
package ru.virgil.security

import org.springframework.boot.context.properties.ConfigurationProperties


@ConfigurationProperties(prefix = "security")
data class SecurityProperties(
    /**
     * По этим путям можно будет заходить анонимно
     */
    var anonymousPaths: List<String> = ArrayList(),
    /**
     * Использовать для сессий заголовок X-Auth-Token вместо Cookies
     */
    var useXAuthToken: Boolean = false,
    /**
     * Отключить редиректы после успешной авторизации
     */
    var disablePostAuthRedirect: Boolean = false,
)
